File: db/migrate/3_create_customers.rb
class CreateCustomers < ActiveRecord::Migration
  def change
    create_table :customers do |t|
      t.string :name
      t.string :phone
      t.string :email
      t.datetime :date_on_waiting_list
      t.integer :status
      t.boolean :local
      t.string :location
      t.string :credit
      t.text :want
      t.text :notes
      t.references :shop, index: true
      t.references :artist, index: true

      t.timestamps
    end
  end
end

File: migrations/3_create_customers.js
exports.up = function (knex) {
  return knex.schema
    .createTable('customers', function(t) {
      t.increments('id');
      t.integer('artist_id').notNullable().references('id').inTable('artists');
      t.string('name').notNullable();
      t.string('phone').notNullable();
      t.string('email');
      t.dateTime('dateonwaitinglist');
      t.integer('status');
      t.boolean('local');
      t.string('location');
      t.string('credit');
      t.text('want');
      t.text('notes');
    });
};

exports.down = function (knex) {
  return knex.schmea
    .dropTable('customers');
};
